### PR TITLE
chore: use var instead of def for extern variables

### PR DESCRIPTION
**Pull request by i10416**
As far as I know,  extern var can bind c variable.
Besides, variables are not read only, thus I think `var` is preferable to method.
1. setting `optind` to 1 allows users to traverse argv again
2. setting `opterr` to 0 disables error log output

### CODE REVIEW

**Comment by i10416** [comment]
As a side note, I think this issue can be closed. Scala Native has been able to bind c variables with extern var.  

https://github.com/scala-native/scala-native/issues/1204

**Comment by LeeTibbert** [comment]
i10416   Good catch!  Thank you. 

Looks like assignment to an extern var can be done since PR #1348 which was committed
over 4 years ago.  Change comes slowly to a code base.

This change Looks Good To Me (LGTM).

A couple of things outside the scope of this PR, but motivated by it:

1) I will create an issue for errno (gai_errno?) and kin, which could gain
    spend a few less cycles in execution using this idiom.

2) `getopt.scala`, introduced in PR #1348 is no longer part of POSIX
     2018 and should be marked in the file and in posixlib.rst as
     no longer part of POXIX and moved to `unistd.scala`.

    getopt.scala has been using the extern var idiom since its creation.
    I believe it is tested even today in each CI run.

**Comment by ekrich** [comment]
I believe issue #1204 is about binding directly to C pre-processor constants and functions which takes a helper function currently.

**Comment by i10416** [comment]
> I believe issue https://github.com/scala-native/scala-native/issues/1204 is about binding directly to C pre-processor constants and functions which takes a helper function currently.

Yeah, c preprocessor constants like `#define MY_VAR 42` still need wrapper function.

e.g.
```
long scalanative__posix_version() { return _POSIX_VERSION; }
```

However,  normal c variables can be bind to `var`.

The link in https://github.com/scala-native/scala-native/issues/1204 points to the code in [the comment](https://gitter.im/scala-native/scala-native?at=5ab7a2cac574b1aa3e2e7ed7) in gitter.

main.c
```
extern char _binary_main_md_start;
char* main_md_start() { return & _binary_main_md_start; }
```

```
@extern
object cdata {
 def _binary_main_md_start : Ptr[ CString ] = extern // segfault
 def main_md_start : Ptr[ CChar ] = extern // works as expected
}
```

He complained that he wanted to bind `_binary_main_md_start` directly, not by `main_md_start` wrapper function.
And I assume that it is possible by `var _binary_main_md_start: Char = extern`.

In fact, scala-native cannot bind `extern const` and c preprocessor values defined by `#DEFINE MYVAR ...` to Scala `val` now, but I feel that title "support binding to top-level constants" a bit misleading for linked comment. Now that  scala-native supports extern var, I think #1204
 can be closed( and it is reasonable to create a dedicated issue for `extern const`s and cpreprocessor constants, isn't it?)

**Comment by i10416** [comment]
TODO
- getopt stuffs in getopt.scala is no longer POSIX and should be removed.
- announce breaking change in 0.5.0.md

**Review by WojciechMazur — APPROVED** [review]

File: posixlib/src/main/resources/scala-native/unistd.c [hunk 1]
```diff
@@ -7,19 +7,14 @@
...
-
-int scalanative_optind() { return optind; }
-
-int scalanative_optopt() { return optopt; }
+// https://man7.org/linux/man-pages/man7/environ.7.html
+// Historically and by standard, environ must be declared in the
+// user program. However, as a (nonstandard) programmer
+// convenience, environ is declared in the header file <unistd.h> if
+// the _GNU_SOURCE feature test macro is defined
+#if #defined(_GNU_SOURCE)
```
Comment (i10416) [inline posixlib/src/main/resources/scala-native/unistd.c:15]:
```suggestion
#if !defined(_GNU_SOURCE)
```

**Comment by i10416** [comment]
Could someone merge this?

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**